File: src/main/java/com/okccc/realtime/utils/PropertiesUtil.java
package com.okccc.realtime.utils;

import java.io.IOException;
import java.util.Properties;

/**
 * Author: okccc
 * Date: 2022/1/29 3:36 下午
 * Desc: 读取配置文件的工具类
 */
public class PropertiesUtil {

    public static Properties load(String fileName) {
        // 创建Properties对象
        Properties prop = new Properties();
        try {
            // ClassLoader陷阱：代码打包提交到linux后会读不到resources资源文件导致空指针异常
            // 代码报错不要慌,Caused by: XxxException下面找到自己写的那几行代码,定位问题根源
//            prop.load(ClassLoader.getSystemClassLoader().getResourceAsStream(fileName));
            prop.load(Thread.currentThread().getContextClassLoader().getResourceAsStream(fileName));
        } catch (IOException e) {
            e.printStackTrace();
        }
        return prop;
    }

    public static void main(String[] args) {
        Properties prop = PropertiesUtil.load("config.properties");
        System.out.println(prop.getProperty("bootstrap.servers"));
    }
}

File: src/main/scala/com/okccc/util/PropertiesUtil.scala
package com.okccc.util

import java.io.InputStreamReader
import java.nio.charset.StandardCharsets
import java.util.Properties

/**
 * Author: okccc
 * Date: 2020/12/12 17:16
 * Desc: 读取配置文件的工具类
 */
object PropertiesUtil {

  def load(filename: String): Properties = {
    // 创建Properties对象
    val prop: Properties = new Properties()
    // 加载resources配置文件
    prop.load(Thread.currentThread().getContextClassLoader.getResourceAsStream(filename))
    prop
  }

  def main(args: Array[String]): Unit = {
    val prop: Properties = PropertiesUtil.load("config.properties")
    val value: String = prop.getProperty("bootstrap.servers")
    println(value)
  }

}
